packages/vite-plugin-cloudflare/src/export-types.ts

```diff
@@ -0,0 +1,88 @@
+import assert from "node:assert";
+import { debuglog } from "./utils";
+import type { CloudflareDevEnvironment } from "./cloudflare-environment";
+import type { WorkersResolvedConfig } from "./plugin-config";
+import type { Miniflare } from "miniflare";
+import type * as vite from "vite";
+
+export type ExportTypes = Record<
+	string,
+	"DurableObject" | "WorkerEntrypoint" | "WorkflowEntrypoint"
+>;
+
+/**
+ * Fetches the export types for all Workers and returns them in a Map
+ */
+export async function getCurrentWorkerNameToExportTypesMap(
+	resolvedPluginConfig: WorkersResolvedConfig,
+	viteDevServer: vite.ViteDevServer,
+	miniflare: Miniflare
+): Promise<Map<string, ExportTypes>> {
+	const results = await Promise.all(
+		[...resolvedPluginConfig.environmentNameToWorkerMap].map(
+			async ([environmentName, worker]) => {
+				debuglog(`Fetching export types for worker "${worker.config.name}"`);
+				const exportTypes = await (
+					viteDevServer.environments[
+						environmentName
+					] as CloudflareDevEnvironment
```

Comment:
Would it be worth introducing a smal helper like `getCloudflareDevEnvironment` that throws if the environment is not an instance of `CloudflareDevEnvironment`?